src/extension.ts

```diff
@@ -71,6 +71,16 @@ async function workWithServer(verb: string, callback: ServerFunction): Promise<v
     const fileName = basename(editor.document.uri.fsPath);
     const fileContent = editor.document.getText();
     let fileProject: string;
+    // The server would not show errors when validating/lemmatising the file if
+    // it doesn't have the `.atf` extension.  We could work around this
+    // limitation, but not setting the extension correctly is likely an error
+    // anyway, so we enforce it to when submitting tasks to the server.
+    if (!fileName.match(/\.atf$/)) {
```

Comment:
Is it a bit more readable to use [`endsWith`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith)?
```suggestion
    if (!fileName.endsWith(".atf")) {
```